pkg/resource/resource.go

```diff
@@ -197,119 +135,14 @@ func CreateFromYAML(yamlFile []byte,
-		obj.SetNamespace(namespace)
-	}
-
-	yamlKind := obj.GetKind()
-	yamlName := obj.GetName()
-	yamlNamespace := obj.GetNamespace()
-	metricValue := 0
-	defer func() {
-		metricsClient.SetCompletedKind(name, yamlKind, yamlName, yamlNamespace, metricValue)
-	}()
-
-	// We used this for predicate filtering, we're watching a lot of
-	// API Objects we want to ignore all objects that do not have this
-	// label.
-	if err = filter.SetLabel(obj); err != nil {
-		return fmt.Errorf("could not set label: %w", err)
-	}
-	// kernel affinity related attributes only set if there is an
-	// annotation specialresource.openshift.io/kernel-affine: true
-	if kernel.IsObjectAffine(obj) {
-		if err = kernel.SetAffineAttributes(obj, kernelFullVersion, operatingSystemMajorMinor); err != nil {
-			return fmt.Errorf("cannot set kernel affine attributes: %w", err)
-		}
-	}
-
-	// Add nodeSelector terms defined for the specialresource CR to the object
-	// we do not want to spread HW enablement stacks on all nodes
-	if err = SetNodeSelectorTerms(obj, nodeSelector); err != nil {
-		return fmt.Errorf("setting NodeSelectorTerms failed: %w", err)
-	}
-
-	// We are only building a driver-container if we cannot pull the image
-	// We are asuming that vendors provide pre compiled DriverContainers
-	// If err == nil, build a new container, if err != nil skip it
-	if err = rebuildDriverContainer(obj); err != nil {
-		log.Info("Skipping building driver-container", "Name", obj.GetName())
-		return nil
-	}
-
-	// Callbacks before CRUD will update the manifests
-	if err = BeforeCRUD(obj, owner); err != nil {
-		return fmt.Errorf("before CRUD hooks failed: %w", err)
-	}
-	// Create Update Delete Patch resources
-	err = CRUD(obj, releaseInstalled, owner, name, namespace)
-	if err != nil {
-		if strings.Contains(err.Error(), "failed calling webhook") {
-			return fmt.Errorf("webhook not ready, requeue: %w", err)
-		}
-
-		return fmt.Errorf("CRUD exited non-zero on Object: %+v: %w", obj, err)
-	}
-
-	// Callbacks after CRUD will wait for ressource and check status
-	if err = AfterCRUD(obj, namespace); err != nil {
-		return fmt.Errorf("after CRUD hooks failed: %w", err)
-	}
-
-	sendNodesMetrics(obj, name)
-
-	metricValue = 1
-	return nil
-}
-
-func IsOneTimer(obj *unstructured.Unstructured) (bool, error) {
-
-	// We are not recreating Pods that have restartPolicy: Never
-	if obj.GetKind() == "Pod" {
-		restartPolicy, found, err := unstructured.NestedString(obj.Object, "spec", "restartPolicy")
-		if err != nil || !found {
-			return false, fmt.Errorf("error or restartPolicy not found: %w", err)
-		}
-
-		if restartPolicy == "Never" {
-			return true, nil
-		}
-	}
-
-	return false, nil
-}
-
 // CRUD Create Update Delete Resource
-func CRUD(obj *unstructured.Unstructured, releaseInstalled bool, owner v1.Object, name string, namespace string) error {
+func (c *creator) CRUD(obj *unstructured.Unstructured, releaseInstalled bool, owner v1.Object, name string, namespace string) error {
```

Comment:
lets make it a private function

Reply:
We agreed to address this with https://issues.redhat.com/browse/MGMT-8567